interfaces/cython/cantera/examples/onedim/UnscaleVariances.py

```diff
@@ -0,0 +1,655 @@
+# -*- coding: utf-8 -*-
+
+#################################################################################################
+## Copyright (C) 2017 The University Of Dayton. All Rights Reserved.
+##
+## No part of this program may be photocopied, transferred, or otherwise reproduced in machine
+## or human readable form without the prior written consent of The University Of Dayton.
+#################################################################################################
+#################################################################################################
+## THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESSED OR IMPLIED WARRANTIES,
+## INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+## FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+## UNIVERSITY OF DAYTON OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+## INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+## LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+## OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+## LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+## NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+## EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.? THE UNIVERSITY OF DAYTON
+## HAS NO OBLIGATION TO SUPPORT THE SOFTWARE.
+#################################################################################################
+
+"""
+Please cite this work as follows:
+Briones, A.M., Olding, R., Sykes, J.P., Rankin, B.A., McDevitt, K., Heyne, J.S., 
+"Combustion Modeling Software Development, Verification and Validation," Proceedings of the 2018 ASME Power & Energy Conference,
+PowerEnergy2018-7433, Lake Buena Vista, FL. 
+"""
+
+import sys
+import getopt
+import numpy as np
+from scipy.interpolate import interp1d
+from matplotlib import cm
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import axes3d,Axes3D
+from multiprocessing import Process, Pipe
+from collections import OrderedDict
+
+# UDRI imports
+from table_def import TableDef
+from Utils import PrintFlush
+from Graphics import PlotDlg
+
+# Input FileHandler class - responsible for reading input file 
+class InputFileHandler():
+
+    def __init__(self, fileName):
+        self._fileName = fileName
+        self._numGridpoints = 0     # num fmean
+        self._numFlamelets = 0      # num cmean
+        self._numFvars = 0          # num fvar 
+        self._numVars = 0           # num variables 
+        self._numCvars = 1          # num of cvar
+        self._numSpecies = 0        # num of species
+        self._numQoi = 0            # num QOI
+        self._columnHeader = "none" # column headings
+        self._data = []             # table data
+        self._input = open(fileName,'r')

```

Comment:
Please do not leave open file handles like this. The scope of the input file handle should be limited such that it can fit in a context block:

    with open (fileName, 'r'):
        # read the file

If needed, you can store the file name itself as an instance variable, or the data read from the file. In addition, variables should be named with underscores, rather than camelCase:

    self._fileName -> self._file_name or self._filename